packages/flutter/lib/src/widgets/media_query.dart

```diff
@@ -266,7 +286,40 @@ class MediaQueryData {
   ///
   ///  * [MediaQuery.textScaleFactorOf], a method to find and depend on the
   ///    textScaleFactor defined for a [BuildContext].
-  final double textScaleFactor;
+  @Deprecated(
+    'Use textScaler instead. '
+    'This enables non-linear accessibility font scaling on Android 14. '
+    'This feature was deprecated after [TBD].',
+  )
+  double get textScaleFactor => textScaler.textScaleFactor;
+  final double _textScaleFactor;
+
+  /// The font scaling strategy to use for laying out textual contents.
+  ///
+  /// If this [MediaQueryData] is created by the [MediaQueryData.fromView]
+  /// constructor, this property reflects the platform's preferred text scaling
+  /// strategy, and may change as the user changes the scaling factor in the
+  /// operating system's accessibility settings.
+  ///
+  /// See also:
+  ///
+  ///  * [MediaQuery.textScalerOf], a method to find and depend on the
+  ///    [textScaler] defined for a [BuildContext].
+  ///  * [TextPainter], a class that lays out and paints text.
+  TextScaler get textScaler {
+    // The constructor was called with an explicitly specified textScaler value,
+    // we assume the caller is unmigrated and ignore _textScaleFactor.
+    if (!identical(TextScaler.noScaling, _textScaler)) {
```

Comment:
This code (and comment) looks like it could be somewhat simplified if we use `null` for the property that isn't used... At least I wouldn't have to remember what we use as the default values...

Reply:
I'm a bit concerned about changing the constructor signature and make something nullable just for migrating old code. I used to use a private `TextScaler` class as the default but I think that's a bit of an overkill and replaced that with comments.

Reply:
What concerns do you have about making these nullable? It is not a breaking change and gives you a clearer signal that something is unused (instead of just assuming its unused if it has a default value, that could also be a valid value).

Reply:
Ah you mean keeping the signature `MediaQuery({ TextScaler? textScaler })` even after everyone's migrated?  I can't really think of issues with that other than it feels a bit strange to have `null` equals to `TextScaler.noScaling`. I guess I can bring back the private TextScaler class.

Reply:
Once we remove textScaleFactor, I would make `textScaler` non-nullable - which also shouldn't be breaking since we previously enforced with an assert that one of them is set. Therefore, if textScaleFactor is gone, textScaler must be set.